src/presentation/relevant/gift/gift-index.presenter.tsx

```diff
@@ -0,0 +1,81 @@
+import type { FC } from 'react';
+import { GiftItem } from './component/gift-item/gift-item.presenter';
+import type { GiftItemProps } from './component/gift-item/gift-item.presenter';
+
+import type { Gift } from '@/model/gift/gift.model';
+import { Card } from '@/presentation/primitive/component/card/card.presenter';
+
+export type GiftIndexProps = {
+  gifts: Array<Pick<Gift, 'id' | 'iconUrl' | 'name' | 'price' | 'remaining'>> | undefined | null;
+  consumablePoint?: number | undefined | null;
+  onExchange: GiftItemProps['onExchange'];
+};
+
+const resolvePlaceholderMessage = (gifts: GiftIndexProps['gifts']): string => {
+  if (gifts === undefined) {
+    return '読み込み中です...';
+  }
+  if (gifts === null) {
+    return '読み込み中にエラーが発生しました';
+  }
+  if (gifts.length < 1) {
```

Comment:
仮に`-1`とかが返ってきたのを例外として検知できないと怖いので、

```suggestion
  if (gifts.length == 0) {
```

でいいんじゃないかと思います。